nx/lib/nx/backend.ex

```diff
@@ -234,10 +249,85 @@ defmodule Nx.Backend do
     chunk_each(dim - 1, rest, [doc | acc], limit, fun)
   end
 
-  defp inspect_value(%Complex{} = val), do: Complex.to_string(val)
-  defp inspect_value(integer) when is_integer(integer), do: Integer.to_string(integer)
-  defp inspect_value(float) when is_float(float), do: Float.to_string(float)
-  defp inspect_value(:neg_infinity), do: "-Inf"
-  defp inspect_value(:infinity), do: "Inf"
-  defp inspect_value(:nan), do: "NaN"
+  defp inspect_value(float, precision) when is_float(float) and is_integer(precision) do
+    float_to_string(float, precision)
+  end
+
+  defp inspect_value(float, nil) when is_float(float), do: Float.to_string(float)
```

Comment:
Nitpick. For consistency with complex_to_string, move the precision handling inside?

```suggestion
  defp inspect_value(float, precision) when is_float(float), do: float_to_string(float, precision)
```